modules/internal/ChapelDomain.chpl

```diff
@@ -1341,17 +1339,25 @@ module ChapelDomain {
           compilerError("sparse arrays of non-default-initializable types are not currently supported");
         }
       }
+    }
 
-      if chpl_warnUnstable then
-        if this.isRectangular() && this.stridable then
-          if rank == 1 {
-            if this.stride < 0 then
-              warning("arrays with negatively strided dimensions are not particularly stable");
-          } else {
-            for s in this.stride do
-              if s < 0 then
-                warning("arrays with negatively strided dimensions are not particularly stable");
+    proc chpl_checkNegativeStride() /*private*/ {
+      // add compile-time checks once ranges allow that
+      if this.isRectangular() && this.stridable {
+        for s in chpl__tuplify(this.stride) do
+          if s < 0 {
+            warning("arrays with negatively strided dimensions are deprecated;"
```

Comment:
Rather than 'deprecated', I'd probably say 'not currently supported', since I think our intention is to improve support for them over time, and 'deprecated' suggests that they're going away without any recourse.